tests/end_to_end_tests/run_in_remote_machine/ec2.py

```diff
@@ -0,0 +1,361 @@
+import dataclasses
+import logging
+import os
+import pathlib as pl
+import shlex
+import time
+import random
+import json
+from typing import List, Dict, Optional
+
+"""
+This module defines main logic that is responsible for manipulating with ec2 instances to run end to end tests
+inside them.
+"""
+
+logger = logging.getLogger(__name__)
+
+# All the instances created by this script will use this string in the name.
+INSTANCE_NAME_STRING = "automated-agent-tests-"
+assert "-tests-" in INSTANCE_NAME_STRING
```

Comment:
Why have this assertion?

Reply:
Yeah, I think that may not be needed. I just copied it from the current [ami tests,](https://github.com/scalyr/scalyr-agent-2/blob/f5d4602cec828c11795bdd840d3f0830cd925a97/tests/ami/packages_sanity_tests.py#L242) as some other ec2 related things :)